ui/src/palette_visualizer.rs

```diff
@@ -0,0 +1,132 @@
+use crate::ui_traits::{UiTool, View};
+use egui::Color32;
+use egui_extras::{Size, TableBuilder};
+use emu::{
+    ppu::{BG_PALETTE_ADDRESS, OBJ_PALETTE_ADDRESS},
+    render::palette_color::{PaletteColor, PaletteType},
+};
+use std::sync::{Arc, Mutex};
+
+use emu::{arm7tdmi::Arm7tdmi, gba::Gba};
+
+pub struct PaletteVisualizer {
+    gba: Arc<Mutex<Gba<Arm7tdmi>>>,
```

Comment:
Maybe this should be `Arc<Mutex<Gba<Cpu>>>` or similar

Reply:
Yes, but maybe `PaletteVisualizer` is a only Arm7tdmi tool. In other case we need to put the `InternalMemory` outside cpu or we need to implement in the trait a method for get it.